libraries/AP_Hygrometer/AP_Hygrometer_UAVCAN.cpp

```diff
@@ -0,0 +1,177 @@
+
+    if (hygrometer_listener_res < 0) {
+        AP_HAL::panic("UAVCAN hygrometer subscriber start problem\n");
+    }
+}
+
+AP_Hygrometer_Backend* AP_Hygrometer_UAVCAN::probe(AP_Hygrometer::Param &_params)
+{
+    WITH_SEMAPHORE(_sem_registry);
+
+    AP_Hygrometer_UAVCAN* backend = nullptr;
+
+    for (uint8_t i = 0; i < AP_HYGROMETER_MAX_SENSORS; i++) {
+        if (_detected_modules[i].driver == nullptr && _detected_modules[i].ap_uavcan != nullptr) {
+            backend = new AP_Hygrometer_UAVCAN(_params);
+            if (backend == nullptr) {
+                AP::can().log_text(AP_CANManager::LOG_INFO,
+                                   LOG_TAG,
+                                   "Failed register UAVCAN Hygrometer Node %d on Bus %d\n",
+                                   _detected_modules[i].node_id,
+                                   _detected_modules[i].ap_uavcan->get_driver_index());
+            } else {
+                _detected_modules[i].driver = backend;
+                AP::can().log_text(AP_CANManager::LOG_INFO,
+                                   LOG_TAG,
+                                   "Registered UAVCAN Hygrometer Node %d on Bus %d\n",
+                                   _detected_modules[i].node_id,
+                                   _detected_modules[i].ap_uavcan->get_driver_index());
+            }
+            break;
+        }
+    }
+    return backend;
+}
+
+AP_Hygrometer_UAVCAN* AP_Hygrometer_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_uavcan, uint8_t node_id)
+{
+    if (ap_uavcan == nullptr) {
+        return nullptr;
+    }
+
+    for (uint8_t i = 0; i < AP_HYGROMETER_MAX_SENSORS; i++) {
+        if (_detected_modules[i].driver != nullptr &&
+            _detected_modules[i].ap_uavcan == ap_uavcan &&
+            _detected_modules[i].node_id == node_id ) {
+            return _detected_modules[i].driver;
+        }
+    }
+
+    bool detected = false;
+    for (uint8_t i = 0; i < AP_HYGROMETER_MAX_SENSORS; i++) {
+        if (_detected_modules[i].ap_uavcan == ap_uavcan && _detected_modules[i].node_id == node_id) {
+            detected = true;
+            break;
+        }
+    }
+
+    if (!detected) {
+        for (uint8_t i = 0; i < AP_HYGROMETER_MAX_SENSORS; i++) {
+            if (_detected_modules[i].ap_uavcan == nullptr) {
+                _detected_modules[i].ap_uavcan = ap_uavcan;
+                _detected_modules[i].node_id = node_id;
+                break;
+            }
+        }
+    }
+
+    return nullptr;
+}
+
+void AP_Hygrometer_UAVCAN::handle_hygrometer(AP_UAVCAN* ap_uavcan, uint8_t node_id, const HygrometerCb &cb)
+{
+    WITH_SEMAPHORE(_sem_registry);
+
+    AP_Hygrometer_UAVCAN* driver = get_uavcan_backend(ap_uavcan, node_id);
+
+    if (driver != nullptr) {
+        WITH_SEMAPHORE(driver->_sem_hygrometer);
+        driver->_humidity = cb.msg->humidity;
+        if (!isnan(cb.msg->temperature))  {
+            driver->_temperature = cb.msg->temperature - C_TO_KELVIN;
+        }
```

Comment:
```suggestion
        driver->_temperature = cb.msg->temperature;
        if (!isnan(driver->_temperature)) {
            driver->_temperature -= C_TO_KELVIN;
        }
```

Otherwise temperature flatlines which is not helpful.